### ISSUE TITLE
⌨ Screening and assessment missing semantic row headers

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#12207]
## ⌨ Accessibility Issue

The screening and assessment table has rows for each skill as well as the education requirement. These act as row headers in addition to the column headers (assessment steps). However, they are only visually displayed as row headers by giving it bold font.

We need to semantically indicate these are row headers as well for ATs.

### WCAG 2.1 Criteria

- [H51: Using table markup to present tabular information](https://www.w3.org/TR/WCAG20-TECHS/H51)
- [H63: Using the scope attribute to associate header cells and data cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H63)

## 🦋 Expected Behaviour

Row headers can be determined both visually and programmatically.

## 📋 Steps to Reproduce

1. Login as admin `admin@test.com`
2. Navigate to an application page on the admin `/admin/candidates/{candidateId}/application
3. Inspect the tables accessibility tree
4. Observe the row headers are plain `td`'s

## 📸 Screenshot

![2024-12-05_08-38](https://github.com/user-attachments/assets/e6bfb2e1-b695-46b3-b3ca-128de0ead070)

## 🙋‍♀️ Proposed Solution

Update the first column cells to the `th[scope=row]`

See: [Table with header cells in the top row and first column](https://www.w3.org/WAI/tutorials/tables/two-headers/)

## ✅ Acceptance Criteria

- [ ] Row headers are programtically determined with the proper scope
- [ ] No visual changes to the table

**Comment by tristan-orourke** [comment]
May be more difficult with react-table 😢
